docs/cody/clients/model-configuration.mdx

```diff
@@ -515,4 +306,221 @@ To enable StarCoder go to **Site admin > Site configuration** (`/site-admin/conf
+
+## Advanced Configuration
+
+For most administrators, relying on the LLM models made available by Cody Gateway is sufficient. However, if even more customization is required, you can configure your own LLM providers and models.
+
+<Callout type="warning">Defining your own LLM providers and models is an advanced use-case and requires care to get the correct results. It also may bypass protections to ensure compatibility between your Sourcegraph instance and LLMs. If you need help contact your Sourcegraph account executive.</Callout>
+
+### Overview
+
+The `"modelConfiguration"` section exposes two fields `"providerOverrides"` and `"modelOverrides"`. These may override any Sourcegraph-supplied data, or simply introduce new ones entirely.
+
+### Provider Configuration
+
+A "provider" is a way to organize LLM models. Typically a provider would be referring to the company that produced the model. Or the specific API/service being used to access the model. But conceptually, it's just a namespace.
+
+By defining a provider override in your Sourcegraph site configuration, you are introducing a new namespace to contain models. Or customize the existing provider namespace supplied by Sourcegraph. (e.g. all `"anthropic"` models.)
+
+The following configuration shippet defines a single provider override with the ID `"anthropic"`.
+
+```json
+"modelConfiguration": {
+  // Do not use any Sourcegraph-supplied models.
+  "sourcegraph": null,
+
+  // Define a provider for "anthropic".
+  "providerOverrides": [
+    {
+      "id": "anthropic",
+      "displayName": "Anthropic models, sent directly to anthropic.com",
+
+      // The server-side config section defines how this provider operates.
+      "serverSideConfig": {
+        "type": "anthropic",
+        "accessToken": "sk-ant-api03-xxxxxxxxx",
+        "endpoint": "https://api.anthropic.com/v1/messages"
+      },
+
+      // The default model configuration provides defaults for all LLM
+      // models using this provider.
+      "defaultModelConfig": {
+        "capabilities": [
+          "chat",
+          "autocomplete"
+        ],
+        "contextWindow": {
+          "maxInputTokens": 10000,
+          "maxOutputTokens": 4000
+        },
+        "category": "balanced",
+        "status": "stable"
+      }
+    }
+  ],
+  ...
+}
+```
+
+**Server-side Configuration**
+
+The most important part of a provider's configuration is the `"serverSideConfig"` field. That defines how the LLM model's should be invoked, i.e. which external service or API will be called to serve LLM requests.
+
+In the example, the `"type"` field was `"anthropic"`. Meaning that any interactions using the `"anthropic"` provider would be sent directly to Anthropic, at the supplied `endpoint` URL using the given `accessToken`.
+
+However, Sourcegraph supports several different types of LLM API providers natively. The current set of supported LLM API providers is:
+
+| Provider type        | Description  |
```

Comment:
Noting that we still do not document the correct options for `providerOverrides.[].id` anywhere in this doc. I think it would be good to include this information here in the same table, since I assume there is 1 to 1 mapping between `providerOverrides.[].id` and `providerOverrides.[]serverSideConfig.type`

Reply:
Agreed and described in more detail here the work we should be doing: https://linear.app/sourcegraph/issue/PRIME-526/document-accepted-values-for-provideroverrides%5B%5Did-in-new